jigsaw-game/common/tessellation.py

```diff
@@ -0,0 +1,147 @@
+from collections.abc import Generator
+from operator import itemgetter
+from random import shuffle
+from typing import Callable
+
+from numpy import (
+    arange,
+    array,
+    column_stack,
+    hstack,
+    meshgrid,
+    ones,
+    random,
+    sqrt,
+    vstack,
+    zeros,
+)
```

Comment:
I think this would be better as just `import numpy as np`. I don't think I have ever seen importing so many functions from numpy before. 👀

Reply:
We actually aren't using that many functions from numpy surprised that we didn't get something from flake8 or black. At this point you may be noticing that I don't care to follow the trend when I don't have to and the trend to use `np.` for all `numpy` is imports superfluous. We do import more so many functions from numpy that it does use the multiline import syntax I am willing to concede to the trend

Reply:
Why would you not want to use common trends? They are established for a reason; in particular to avoid name clashes. If you think this is something that won't affect the project from the reviewers, then it should be fine to keep. I would advise though to stick to these "trends" as often as possible.

Reply:
I am fully aware of name clashing, but I hardly see that happening in this project, at best if we import `math.sqrt` alongside `numpy.sqrt` or the admittedly more real issues we could get from `scipy`, but we should be decisive. to use the correct functions from both `numpy` and `scipy`  and the types/parameters are cross compatible so no issues there I think.  But I do appreciate you bringing it up